util/dvsim/Deploy.py

```diff
@@ -309,24 +311,50 @@ def get_status(self):
             self.link_odir()
             del self.process
 
+    @staticmethod
+    def increment_timer():
+        # sub function that increments with overflow = 60
+        def _incr_ovf_60(val):
+            if val >= 59:
+                val = 0
+                return val, True
+            else:
+                val += 1
+                return val, False
+
+        incr_hh = False
+        Deploy.ss, incr_mm = _incr_ovf_60(Deploy.ss)
+        if incr_mm: Deploy.mm, incr_hh = _incr_ovf_60(Deploy.mm)
+        if incr_hh: Deploy.hh += 1
+
     @staticmethod
     def deploy(items):
         dispatched_items = []
         queued_items = []
 
-        if Deploy.print_legend:
-            # Print legend once at the start of the run.
-            log.info("[legend]: [Q: queued, D: dispatched, " + \
-                     "P: passed, F: failed, K: killed, T: total]")
-            Deploy.print_legend = False
-
-        def get_etime():
-            # Convert num_secs to hh:mm:ss
-            hh = Deploy.num_secs // (3600)
-            ss = Deploy.num_secs % (3600)
-            mm = ss // 60
-            ss %= 60
-            return "%02i:%02i:%02i" % (hh, mm, ss)
+        # Print timer val in hh:mm:ss.
+        def get_timer_val():
+            return "%02i:%02i:%02i" % (Deploy.hh, Deploy.mm, Deploy.ss)
+
+        # Check if elapsed time has reached the next print interval.
+        def has_print_interval_reached():
+            # Deploy.print_interval is expected to be < 1 hour.
+            return (((Deploy.mm * 60 + Deploy.ss) %
+                     Deploy.print_interval) == 0)
+
+        # Move items from queued_items to dispatch_items list depending on
+        # available slots.
+        def dequeue_items(dispatch_items, queued_items):
+            num_slots = Deploy.max_parallel - Deploy.dispatch_counter
+            if num_slots > 0:
+                if len(queued_items) > num_slots:
+                    dispatch_items(queued_items[0:num_slots])
+                    dispatched_items.extend(queued_items[0:num_slots])
+                    queued_items = queued_items[num_slots:]
+                else:
+                    dispatch_items(queued_items)
+                    dispatched_items.extend(queued_items)
+                    queued_items = []
```

Comment:
seems like sim can't be finished even all the items are done,
because this kind of clearing list doesn't clear the reference used at line 464
`all_done = (len(queued_items) == 0)`

found some solutions from this link
https://stackoverflow.com/questions/850795/different-ways-of-clearing-lists
```suggestion
                    queued_items.clear()
```

Reply:
Wow, good catch. I will update it.